Add support for reverse-only imports

saddfox opened the pull request:
Allow reverse-only imports by passing `--reverse-only` flag during import process.
Controlled via env variable `REVERSE_ONLY`, defaults to false.

[Nominatim docs](https://nominatim.org/release-docs/latest/admin/Import/#reverse-only-imports)

philipkozeny commented:
Thanks for the PR. can you also please add an automated test case in our ci.yml? Thanks

saddfox commented:
Fixed a crash with db warming when running on a reverse-only import. Somehow it was fine with my local pruned map.

I have added a test case to verify nominatim is running in reverse-only mode. In that case it returns http status code 404 and `{"error":{"code":404,"message":"Reverse-only import does not support forward searching."}}` so I am looking for this exact match.

philipkozeny commented:
Thanks for the great PR!
